chia/full_node/full_node.py

```diff
@@ -716,11 +720,7 @@ async def new_peak(self, request: full_node_protocol.NewPeak, peer: WSChiaConnec
                 await self.short_sync_batch(peer, uint32(0), request.height)
                 return None
 
-            if request.height < curr_peak_height + self.config["sync_blocks_behind_threshold"]:
-                # TODO: We get here if we encountered a heavier peak with a
-                # lower height than ours. We don't seem to handle this case
-                # right now. This ends up requesting the block at *our* peak
-                # height.
+            if curr_peak_height <= request.height < curr_peak_height + self.config["sync_blocks_behind_threshold"]:
```

Comment:
where do we go if the new heavier peak is hight current_peak -1 ?

Reply:
we end up on line 730, calling
```
self._sync_task = asyncio.create_task(self._sync())
```

This is the same behavior as today, just without the detour of sending an invalid request, failing and then falling through to line 730

Reply:
I take it you prefer two comparisons with `and` here as well, right?